lib/chess/cli/render.rb

```diff
@@ -9,43 +9,92 @@ class Render
       include Chess::Core::Constants
 
       def initialize
-        @records_history = nil
+        @current_player = nil
+        @records_history = []
       end
 
       def ask_for_prompt
         puts 'make your move:'
       end
 
-      def header(board)
+      def board_state(board)
         puts "\n"
         puts "\n"
-        puts board
+        puts board_to_string(board)
         puts "\n"
         puts "  \u00bb #{@current_player} \u00ab  ".black.on_red if @current_player == BLACK_TEAM
         puts "  \u00bb #{@current_player} \u00ab  ".light_white.on_light_red if @current_player == WHITE_TEAM
       end
 
-      def current_player(player)
+      def print_records_history
+        history = ''
+        row = 1
+        @records_history.each do |record|
+          history += "#{row}. #{record_to_string(record.white)}   #{record_to_string(record.black)}\n"
+          row += 1
+        end
+
+        puts history
+      end
+
+      def update_current_player(player)
         @current_player = player
       end
 
-      def show_record
-        puts "RECORD\n"
-        line = 0
+      def update_records_history(records_history)
+        @records_history = records_history
+      end
+
+      private
 
-        historic_string = @records_history.map do |play_record|
-          line += 1
-          "#{line}. #{play_record}\n"
+      def board_to_string(board)
+        headers = "    a  b  c  d  e  f  g  h    \n".yellow
+
+        row = headers
+        (MAX_INDEX..MIN_INDEX).step(-1).each do |row_index|
+          row += " #{row_index + 1} ".yellow
+          row_of_cells = board.cells[row_index]
+          row_of_cells.each do |cell|
+            row += cell_to_string(cell)
+          end
+          row += " #{row_index + 1} \n".yellow
         end
 
-        puts historic_string.join
+        row += headers
+      end
 
-        @records_history = nil
+      def cell_to_string(cell)
+        return " #{piece_at(cell)} ".on_light_red if cell.fill_color == WHITE_TEAM
+
+        " #{piece_at(cell)} ".on_red if cell.fill_color == BLACK_TEAM
       end
 
-      def update_records_history(records_history)
-        @records_history = records_history
+      def piece_at(cell)
+        return ' ' unless cell.occupied?
+
+        return PIECE_STRING[cell.occupant.symbol].to_s.white if cell.occupant.team == WHITE_TEAM
+
+        PIECE_STRING[cell.occupant.symbol].to_s.black
+      end
+
+      # rubocop:disable all
```

Comment:
nasty hahah. Is there a way to make this method smaller and more readable?

Reply:
yeah hahahah I was working on it:

```ruby
def record_to_string(record)
  response_string = ''
  return response_string unless record

  response_string += base_move_to_string(record)
  response_string += check_or_mate_to_string(record)
  response_string += draw_to_string(record)

  response_string + castle_to_string(record.castling)
end
```